packages/typespec-client-generator-core/test/package.test.ts

```diff
@@ -3091,17 +3055,429 @@ describe("typespec-client-generator-core: package", () => {
       op test(...Intersected): void;
       `);
       const method = getServiceMethodOfClient(runner.context.experimental_sdkPackage);
-      const documentMethodParam = method.parameters.find(
-        (x) => x.name === "documentTranslateContent"
-      );
+      const documentMethodParam = method.parameters.find((x) => x.name === "document");
       ok(documentMethodParam);
       strictEqual(documentMethodParam.kind, "method");
       const op = method.operation;
       ok(op.bodyParam);
       strictEqual(op.bodyParam.kind, "body");
-      strictEqual(op.bodyParam.name, "documentTranslateContent");
+      strictEqual(op.bodyParam.name, "testRequest");
       deepStrictEqual(op.bodyParam.correspondingMethodParams, [documentMethodParam]);
     });
+
+    it("anonymous model with @body should not be spread", async () => {
+      await runner.compileWithBuiltInService(`
+        op test(@body body: {prop: string}): void;
+        `);
+      const method = getServiceMethodOfClient(runner.context.experimental_sdkPackage);
+      const models = runner.context.experimental_sdkPackage.models;
+      strictEqual(models.length, 1);
+      const model = models.find((x) => x.name === "TestRequest");
+      ok(model);
+      strictEqual(model.usage, UsageFlags.Input);
+
+      strictEqual(method.parameters.length, 2);
+      const param = method.parameters[0];
+      strictEqual(param.kind, "method");
+      strictEqual(param.name, "body");
+      strictEqual(param.optional, false);
+      strictEqual(param.isGeneratedName, false);
+      deepStrictEqual(param.type, model);
+      const contentTypeMethoParam = method.parameters.find((x) => x.name === "contentType");
+      ok(contentTypeMethoParam);
+
+      const op = method.operation;
+      strictEqual(op.parameters.length, 1);
+      ok(
+        op.parameters.find(
+          (x) =>
+            x.kind === "header" &&
+            x.serializedName === "Content-Type" &&
+            x.correspondingMethodParams[0] === contentTypeMethoParam
+        )
+      );
+
+      const bodyParam = op.bodyParam;
+      ok(bodyParam);
+      strictEqual(bodyParam.kind, "body");
+      strictEqual(bodyParam.name, "body");
+      strictEqual(bodyParam.optional, false);
+      strictEqual(bodyParam.isGeneratedName, false);
+      deepStrictEqual(bodyParam.type, model);
+      deepStrictEqual(bodyParam.correspondingMethodParams[0].type, model);
+    });
+
+    it("anonymous model from spread with @bodyRoot should not be spread", async () => {
+      await runner.compileWithBuiltInService(`
+        model Test {
+          prop: string;
+        }
+        op test(@bodyRoot body: {...Test}): void;
+        `);
+      const method = getServiceMethodOfClient(runner.context.experimental_sdkPackage);
+      const models = runner.context.experimental_sdkPackage.models;
```

Comment:
we haven't adopted to getAllOperations for modular yet, I guess we can't leveage the tcgc feature here?

Reply:
yes. emitters that has not adopted to getAllOperations need to deal with the operation by themselves. you could refer to the judgement i used in types.ts.